app/services/transaction_service/transaction.rb

```diff
@@ -369,27 +367,23 @@ def checkout_details(model)
   end
 
   def calculate_commission(total_price, commission_from_seller, minimum_commission)
-    if commission_from_seller.blank? || commission_from_seller == 0
-      Money.new(0, minimum_commission.currency)
-    else
-      commission_by_percentage = total_price * (commission_from_seller / 100.0)
-      (commission_by_percentage > minimum_commission) ? commission_by_percentage : minimum_commission
-    end
+    [(total_price * (commission_from_seller / 100.0) unless commission_from_seller.nil?),
+     (minimum_commission unless minimum_commission.nil? || minimum_commission.zero? ),
+     Money.new(0, total_price.currency)]
+      .compact
+      .max
   end
 
-  def calculate_commission_to_admin(commission_total, fee_total)
-    commission_total - fee_total #we charge from admin, only the sum after paypal fees
+  def calculate_commission_to_admin(commission_total, payment_total, fee_total)
+    # Ensure we never charge more than what the seller received after payment processing fee
+    [commission_total, payment_total - fee_total].min
   end
 
-  def paypal_account_verified?(community_id:, person_id:nil)
-    acc = paypal_accounts_api.get(
-      community_id: community_id,
-      person_id: person_id
-    ).maybe
+  def paypal_account_verified?(community_id:, person_id: nil, settings: Maybe(nil))
```

Comment:
The exact same method is in paypal_helper, should this logic be somehow published through service api?